### PR TITLE
feat: create DOI normalization function

### PR DESCRIPTION
**Pull request by ebezzi**
### Reviewers
**Functional:** 
@MDunitz 

**Readability:** 

Ticket [#1803](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/chanzuckerberg/single-cell-data-portal/1803)

---

## Changes
- Adds a function that returns a normalized DOI for a collection. This DOI doesn't have an URL format and will be used to call the Crossref API to retrieve the metadata.

### CODE REVIEW

File: backend/corpora/common/entities/collection.py [hunk 1]
```diff
@@ -280,3 +281,18 @@ def delete(self):
...
+
+    @staticmethod
+    def _normalize_doi(doi_url) -> str:
+        try:
+            return urlparse(doi_url.strip()).path.strip("/")
+        except Exception:
+            return None
+
+    def get_normalized_doi(self) -> str:
+        """Returns a normalized DOI if it exists, None otherwise"""
```
Comment (MDunitz) [inline backend/corpora/common/entities/collection.py:293]: Could you specify what normalized means in this context?

File: backend/corpora/common/entities/collection.py [hunk 2]
```diff
@@ -280,3 +281,18 @@ def delete(self):
...
+        doi = [link for link in self.links if link.link_type == CollectionLinkType.DOI]
```
Comment (MDunitz) [inline backend/corpora/common/entities/collection.py:294]: Is it every possible for this to return more than one link?

Comment (ebezzi) [reply backend/corpora/common/entities/collection.py:294]: Publications should have a unique DOI so in theory, no. I don't think we're enforcing this though, so maybe we should do that when we add links.

Comment (MDunitz) [reply backend/corpora/common/entities/collection.py:294]: Can you create a ticket to ensue it is enforced? And maybe to double check the db to make sure nothing slipped through?

**Review by MDunitz — APPROVED** [review]
lgtm

**Pull request merged**